test/integration/consul-container/libs/utils/docker.go

```diff
@@ -9,13 +9,28 @@ import (
 	"io"
 	"os"
 	"os/exec"
+	"strings"
+
+	"github.com/hashicorp/go-version"
 )
 
 // DockerExec simply shell out to the docker CLI binary on your host.
 func DockerExec(args []string, stdout io.Writer) error {
 	return cmdExec("docker", "docker", args, stdout, "")
 }
 
+// DockerImageVersion retrieves the value of the org.opencontainers.image.version label from the specified image.
+func DockerImageVersion(imageName string) (*version.Version, error) {
+	var b strings.Builder
+	err := cmdExec("docker", "docker", []string{"image", "inspect", "--format", `{{index .Config.Labels "org.opencontainers.image.version"}}`, imageName}, &b, "")
```

Comment:
I wonder if there's a way to do this without shelling out? We already presumably have access to a Docker API client in order to launch containers.

Totally optional though!

Reply:
I think for now I will leave this as is as I don't want to refactor all docker shell execs to use the library. Also as far as I can tell we don't have access to any docker client directly. Thats all hidden away behind some abstractions.